c/httpserver.c

```diff
@@ -3814,32 +3835,50 @@ void respondWithJsonError(HttpResponse *response, char *error, int statusCode, c
   finishResponse(response);
 }
 
-int streamBinaryForFile(Socket *socket, UnixFile *in){
+#define ENCODE64_SIZE(SZ) (2 + 4 * ((SZ + 2) / 3))
+
+int streamBinaryForFile(Socket *socket, UnixFile *in, bool asB64) {
   int returnCode = 0;
   int reasonCode = 0;
-  char buffer[FILE_STREAM_BUFFER_SIZE+4];
+  int bufferSize = 3*FILE_STREAM_BUFFER_SIZE;
```

Comment:
I'm not aware of if we need to observe a size limit here, but it seems like getting the buffer divisible by 3 by tripling it may be better done by adding the remainder to make it divisible by 3.